apps/emqx_resource/src/emqx_resource_cache.erl

```diff
@@ -0,0 +1,331 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_resource_cache).
+
+%% CRUD APIs
+-export([new/0, write/3, is_exist/1, read/1, erase/1]).
+%% For Config management
+-export([all_ids/0, list_all/0, group_ids/1]).
+%% For health checks etc.
+-export([read_status/1, read_mod/1, read_manager_pid/1]).
+%% Hot-path
+-export([get_runtime/1]).
+
+-include("emqx_resource_runtime.hrl").
+-include_lib("stdlib/include/ms_transform.hrl").
+
+-define(CACHE, ?RESOURCE_CACHE).
+
+-record(connector, {
+    id :: binary(),
+    group :: binary(),
+    manager_pid :: pid(),
+    st_err :: st_err(),
+    config :: term(),
+    cb :: term(),
+    extra = []
+}).
+
+-type chan_key() :: {ConnectorId :: binary(), ChannelID :: binary()}.
```

Comment:
Nit:

```suggestion
-type chan_key() :: {connector_res_id(), action_res_id() | source_res_id()}.
```

(maybe we could redefine the existing `channel_id` to `action_res_id() | source_res_id()`)